core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -219,6 +221,32 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
     }
   }
 
+  override def getEventLogPaths(
+      appId: String,
+      attemptId: String): Seq[Path] = {
```

Comment:
Shouldn't `attemptId` be an `Option[String]` like in other places? Not all cluster managers have attempt IDs.